### ISSUE TITLE
Torchvision-like model zoo

### ISSUE THREAD

**Issue by kdexd** [facebookresearch/detectron2#270]
## 🚀 Feature

How about adding support to initialize standard models (such as Faster-RCNN, Mask-RCNN, RetinaNet etc.) with pre-trained weights officially released with Detectron2 (in `MODEL_ZOO.md`)?

## Motivation

Initializing a model with pretrained weights takes a lot of lines of code:

```python
from detectron2.config import get_cfg
from detectron2.modeling import build_model
from detectron2.checkpoint import DetectionCheckpointer

cfg = get_cfg()
cfg.merge_from_file("config_for_model_i_want_to_use.yaml")

model = build_model(cfg)

# Set model weights as a part of url by looking up MODEL_ZOO.md file.
cfg.MODEL.WEIGHTS = "detectron2://COCODetection/faster_rcnn_R_50_C4_1x/{model_id}/model_final_{commit_sha}.pkl"
DetectionCheckpointer(model).resume_or_load(cfg.MODEL.WEIGHTS, resume=False)
```

I wish to use pre-trained detectors for my project, and detectors are a small part in the whole pipeline. I have my own config system/checkpointer etc. and would want to get this boilerplate code to be reduced.

Apart from a lot of boilerplate code, I need to have the config file downloaded separately, and also keep a track of the URL of model weights (to override `cfg.MODEL.WEIGHTS`).

## Pitch

- `detectron2` as a package should include all the config files. This can be easily done using setuptools. 
- There should be a mapping between config file/model name to S3 URL.

With these two in place, creating a pre-trained model would be much simpler.

To make it fancy, the API can be extended and might look something like:

```python
from detectron2.model_zoo import faster_rcnn_R_50_C4_3x
model = faster_rcnn_R_50_C4_3x(pretrained=True)
```

cc: @jcjohnson

**Comment by kdexd** [comment]
I have opened a (draft) Pull Request to better demonstrate a proof of concept. ^
 cc: @jcjohnson
